### ISSUE TITLE
Force refresh charts to always?

### ISSUE THREAD

**Issue by markmicallef42** [firehol/netdata#423]
Hello! Netdata is great :+1: 

Is it maybe possible to force the charts to be refreshed even when not focused in the browser window? I ask this because every time I change the setting, it changes back when I open the netdata of another node :)

Cheers,

**Comment by ktsaou** [comment]
Nice you like data. Thanks!

This is by design. When I did that I was afraid that people will forget their browsers open in a netdata dashboard. This will have 2 effects:

1. drain laptop batteries (chart refreshes require CPU on the client)
2. consume precious CPU server resources (imagine a situation where 5 guys on your team left the office, but their computers have 2 netdata dashboards each - a lot of CPU load on your servers without reason).

For these reasons, I made this particular setting reset when you refresh the page.

**Comment by markmicallef42** [comment]
Hello, 

Thank you very much for the reply and the explanation. I was just thinking of setting up a monitor purely with netdata on it, and just used for monitoring (no keyboard/mouse connected), but it's pretty hard to do because of how it is.

**Comment by ktsaou** [comment]
In this case you will have custom dashboard. You can set a setting in the HTML to enable this permanently and also reload the dashboard every a few minutes.

Go to http://your.netdata:19999/tv.html and view its source. It is very simple.

**Comment by markmicallef42** [comment]
Perfect. Thank you very much and keep up the great work :+1:

**Comment by RX14** [comment]
This is really annoying, I have three monitors and want to be able to perform commands on terminals and do other things while keeping an eye on netdata. I *always* want it to refresh in the background, and changing it every time I open netdata is a pain. I use a desktop so battery and CPU usage are pretty much a non issue for me, and netdata is so efficient that even leaving it refreshing always, the CPU usage rounds to 0%.

Maybe make this an option to enable/disable saving the preference in the netdata config on the host?

**Comment by haasn** [comment]
+1 on this “feature” being annoying and rather undesired in my case. I will proably end up patching it out of my version of netdata.

> This is by design. When I did that I was afraid that people will forget their browsers open in a netdata dashboard. This will have 2 effects:

That doesn't make sense. These sound like arguments for making “on focus” the default behavior (which it *is*!), and that's fine. What this doesn't relate to is why the setting should get reset if you refresh the page.

What use case are you trying to solve here? In which scenario would somebody

1. **Manually** set the setting to “always” instead of “on focus”
2. *Refresh* the tab but then forget to close it?

The only time this feature would have any measurable positive effect is in exactly that scenario, i.e. a user explicitly refreshes the tab. In your described “5 guys on your team left the office” scenario, would those 5 guys really have manually refreshed all of the netdata tabs before leaving the office?

In summary, all it seems to do is cause harm to people with multi-monitor / tiling wm workflows, with no beneficial effect that I can imagine ever realistically being the case.

Also, if you're worried about excess CPU drain, it might be better to kick long-idle clients or add rate limits server-side. I personally hardly think the miniscule amount of CPU time that netdata consumes is worth worrying about, though.

**Comment by ktsaou** [comment]
ok guys, here it is.
It still not saved, but if you refresh the dashboard, or while you jump from server to server via the my-netdata menu, it remains...

**Comment by justinvvitale** [comment]
Just for others, to get it to update always do this to the URL:
http://IP:19999/#menu_system;theme=slate;help=false;update_always=true

**Comment by racile** [comment]
Netdata v1.20.0-1190-nightly

i'm using multiple screen to monitor multiple nodes at the same time, on the main dashboard the onfocus event problem was solved by the option in:

Settings > Performance > When to refresh the charts? 
Change from **On Focus** to **Always**

when i refresh the page it added the **update_always=true** on the url and it works.
http://my.netdata:19999/#menu_system;theme=slate;help=true;update_always=true

however it doesn't work on:
http://my.netdata:19999/tv.html

I’ve even tried to (still no luck):
http://my.netdata:19999/tv.html#menu_system;theme=slate;help=true;update_always=true

viewing the tv.html(source) I’ve found and done this:

edit(uncomment):
tv.html

before:
// always update the charts, even if focus is lost
// NETDATA.options.current.stop_updates_when_focus_is_lost = false;

after
// always update the charts, even if focus is lost
NETDATA.options.current.stop_updates_when_focus_is_lost = false;

save and restart the netdata service
and voilà!, i hope this one helps.

great monitoring software by the way, we'll be implementing this on all our existing and future nodes and recommending this to all our IT colleagues.

thank you, more power and God Bless.
